src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -263,7 +362,14 @@ static void OnResolve(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t inter
     if (!(flags & kDNSServiceFlagsMoreComing))
     {
         VerifyOrReturn(sdCtx->HasInterface(), sdCtx->Finalize(kDNSServiceErr_BadState));
-        GetAddrInfo(sdCtx);
+
+        // If a resolve was not requested on this context, call GetAddrInfo and set the isResolveRequested flag to true.
+        if (!sdCtx->isResolveRequested)
```

Comment:
I don't understand this logic.  We want to call GetAddrInfo any time we get OnResolve without kDNSServiceFlagsMoreComing, and _only do the GetAddrInfo on the things we didn't do it on yet_.

As in, this isResolveRequested flag needs to be per interface/domain pair, not per-context, no?

Reply:
I misunderstood this. This needs rework and I am going to do that as a follow up.

Reply:
Filed: #32681